### ISSUE TITLE
NETCore.App and AspNetCore.App versions mismatch

### ISSUE THREAD

**Issue by Matheos96** [dotnet/runtime#89070]
I am unsure whether this is a bug or just some confusion, hence why I am reporting this as a "blank issue".

I have noticed that my computer at times automatically installs the latest ASP.NET Core runtime. This in itself is not an issue and I think it is due to this setting being enabled (I now disabled it):
![image](https://github.com/dotnet/runtime/assets/32064310/9de0288f-80cc-4f7d-b318-40be97e92e47)
The problem is the fact that it does **not** install the matching NETCore.App runtime version. When I try to run our application in Rider/Visual Studio, the default runtime resolution finds the latest ASP.NET runtime and picks it, but then obviously cannot find the matching NETCore.App runtime --> shows me a message that I need to update it.

This kind of makes sense, apart from the fact that WSUS did not also update base runtime, but here are my questions:

1. Is it true that the ASP.NET/Desktop runtime version **MUST** find a matching NETCore.App runtime? Can't it fallback to use an older (patch) version if available??
2. Why would this auto update through Windows not install the base runtime too? To my understanding, the ASP.NET/Desktop runtime cannot be used without the base anyway???
3. If my assumption in question 1 is correct, then why would the default behaviour not then be to fall back to the latest runtime which has matching versions installed? For example, I had ASPNetCore.App 6.0.20 and NETCore.App 6.0.19 as latest versions (but older ones available). A possible solution would have been to use version 6.0.19 of both and the app would have run fine. Instead it first seemed to choose ASPNet 6.0.20 and then instantly fail as it did not find the corresponding base runtime

**Comment by huoyaoyuan** [comment]
Sounds like https://github.com/dotnet/runtime/issues/66864#issuecomment-1072961998

**Comment by Matheos96** [comment]
> Sounds like [#66864 (comment)](https://github.com/dotnet/runtime/issues/66864#issuecomment-1072961998)

Yes sounds similar. As mentioned, I decided to disabled the windows update thing, though I would be very keen on also getting answers to my questions regarding whether my assumptions are correct or not.

Assumably it would be sufficient for us to install the required runtimes in our installer and have our customers turn off the windows update option too? To prevent issues like this happening on their machines. Or from the comment in the other thread perhaps you mean that this issue only happens on machines that also have Visual Studio installed? Something which customer PCs probably does not have.
